node/src/components/chain_synchronizer/operations.rs

```diff
@@ -725,12 +781,12 @@ async fn archival_sync(
     // Sync forward until we are at the current version.
     info!("start - sync forward - archival sync");
     let mut most_recent_block = trusted_block;
-    while most_recent_block.header().protocol_version() < chainspec.protocol_config.version {
+    while most_recent_block.header().protocol_version() < ctx.chainspec.protocol_config.version {
         let maybe_fetched_block_with_metadata = fetch_and_store_next::<BlockWithMetadata>(
-            effect_builder,
+            *ctx.effect_builder,
             most_recent_block.header(),
             &trusted_key_block_info,
-            chainspec,
+            ctx.chainspec,
```

Comment:
I wonder whether some of these functions should now be `ChainSyncContext` methods instead?
(Doesn't have to be in this PR; just a thought!)